controllers/storagecluster/storagecluster_controller.go

```diff
@@ -92,6 +92,7 @@ type StorageClusterReconciler struct {
 	recorder           *util.EventReporter
 	OperatorCondition  conditions.Condition
 	IsNoobaaStandalone bool
+	StorageProfiles    ocsv1.StorageProfileList
```

Comment:
I think it would be preferable to have this as a pointer and not a value type. It will enable you to load into it without a need for the value (and the entire substruct to be copied). It will also enable us to verify if it was loaded or not using a `nil` value check

```suggestion
	StorageProfiles    *ocsv1.StorageProfileList
```

Reply:
Removed field